cpp/include/raft/util/reduction.cuh

```diff
@@ -157,11 +157,10 @@ DI std::pair<T, i_t> blockRankedReduce(T val,
     val = values[lane];
     idx = indices[lane];
   } else {
-    // get the min if it is a max op, get the max if it is a min op
-    val = reduce_op(std::numeric_limits<T>::min(), std::numeric_limits<T>::max()) ==
-              std::numeric_limits<T>::min()
-            ? std::numeric_limits<T>::max()
-            : std::numeric_limits<T>::min();
+    // get the lower_bound of the type if it is a max op,
+    // get the upper bound of the type if it is a min op
+    val = reduce_op(lower_bound<T>(), upper_bound<T>()) == lower_bound<T>() ? upper_bound<T>()
```

Comment:
It seems like the identity should be known at compile time rather than evaluating a max/min operation to determine the identity for the given operation.